lib/ecto/changeset.ex

```diff
@@ -313,38 +313,43 @@ defmodule Ecto.Changeset do
   Converts the given `params` into a changeset for `data`
   keeping only the set of `required` and `optional` keys.
   """
-  # TODO: Deprecate this function so it finally emit warnings.
   def cast(data, params, required, optional)
 
-  def cast(_data, %{__struct__: _} = params, _required, _optional) do
+  def cast(data, params, required, optional) do
+    IO.puts :stderr, "warning: `Ecto.Changeset.cast/4` is deprecated, " <>
```

Comment:
We are going to do this only on 2.1 (sorry, I should have added this information to the TODO). So what if we bring the TODO back and comment the warning for now? Everything else is great to go though :)